dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -64,7 +64,7 @@ def show
     configure_caching(@script)
     load_script_level
 
-    if request.path != (canonical_path = build_script_level_path(@script_level))
+    if request.path != (canonical_path = build_script_level_path(@script_level, params))
       canonical_path << "?#{request.query_string}" unless request.query_string.empty?
```

Comment:
I changed approach on this to make all the tests work.  params had a bunch of extra baggage we didn't want to get into the URL.